apps/arweave/src/ar_nonce_limiter_server_worker.erl

```diff
@@ -93,88 +92,86 @@ terminate(_Reason, _State) ->
 %%% Private functions.
 %%%===================================================================
 
-push_update(SessionKey, Session, PrevSessionKey, PrevSession, Output, PartitionUpperBound,
-		Peer, Format, State) ->
-	Update = ar_nonce_limiter_server:make_nonce_limiter_update(
-		SessionKey,
-		Session#vdf_session{
-			upper_bound = PartitionUpperBound,
-			steps = [Output]
-		},
-		true),
-	StepNumber = Session#vdf_session.step_number,
-	case ar_http_iface_client:push_nonce_limiter_update(Peer, Update, Format) of
-		ok ->
-			State;
-		{ok, Response} ->
-			RequestedFormat = Response#nonce_limiter_update_response.format,
-			Postpone = Response#nonce_limiter_update_response.postpone,
-			SessionFound = Response#nonce_limiter_update_response.session_found,
-			RequestedStepNumber = Response#nonce_limiter_update_response.step_number,
-
-			case { 
-					RequestedFormat == Format,
-					Postpone == 0,
-					SessionFound,
-					RequestedStepNumber >= StepNumber - 1
-			} of
-				{false, _, _, _} ->
-					%% Client requested a different payload format
-					?LOG_DEBUG([{event, vdf_client_requested_different_format},
-						{peer, ar_util:format_peer(Peer)},
-						{format, Format}, {requested_format, RequestedFormat}]),
-					push_update(SessionKey, Session, PrevSessionKey, PrevSession,
-							Output, PartitionUpperBound, Peer, RequestedFormat,
-							State#state{ format = RequestedFormat });
-				{true, false, _, _} ->
-					%% Client requested we pause updates
-					Now = os:system_time(second),
-					State#state{ pause_until = Now + Postpone };
-				{true, true, false, _} ->
-					%% Client requested the full session
-					case PrevSession of
-						undefined ->
-							ok;
-						_ ->
-							push_session(PrevSessionKey, PrevSession, Peer, Format)
-					end,
-					push_session(SessionKey, Session, Peer, Format),
-					State;
-				{true, true, true, false} ->
-					%% Client requested missing steps
-					push_session(SessionKey, Session, Peer, Format),
+push_update(SessionKey, StepNumber, Output,
+		PartitionUpperBound, Peer, Format, State) ->
+	Session = ar_nonce_limiter:get_session(SessionKey),
+	Update = ar_nonce_limiter_server:make_partial_nonce_limiter_update(
+		SessionKey, Session, StepNumber, Output, PartitionUpperBound),
+	case Update of
+		not_found -> ok;
```

Comment:
We need to return `State` here.

Reply:
Thanks!

Reply:
Fixed: https://github.com/ArweaveTeam/arweave/pull/498/commits/9818fdfdf7e4a1bb2c87b83cd1f86d82bb1e58d5